pkg/cli/tools/binary_tools.go

```diff
@@ -172,6 +172,7 @@ func DownloadToFolder(filepath string) error {
 	if err != nil {
 		return fmt.Errorf("failed to read file attributes %s: %v", filepath, err)
 	}
+	defer bicepBinary.Close()
```

Comment:
I think this should be moved up to line 169 after where `os.Open` is called. This will ensure that the file is closed as soon as the function returns, even if it returns due to an error.